src/components/custom/array-field/types.ts

```diff
@@ -0,0 +1,29 @@
+import * as Icons from "@lifesg/react-icons";
+import { IYupValidationRule, TFrontendEngineFieldSchema } from "../../frontend-engine/types";
+import { IBaseCustomFieldSchema } from "../types";
+
+export interface IArrayFieldValidationRule extends IYupValidationRule {
+	/** for customising error message when one section is invalid */
+	valid?: boolean | undefined;
+}
+
+export interface IArrayFieldButton {
+	label?: string | undefined;
+	icon?: keyof typeof Icons | undefined;
+}
+
+export interface IArrayFieldRemoveConfirmationModal {
+	title?: string | undefined;
+}
+
+export interface IArrayFieldSchema<V = undefined>
+	extends IBaseCustomFieldSchema<"array-field", V, IArrayFieldValidationRule> {
+	// TODO: introduce unique rule for children of fieldSchema
+	fieldSchema: Record<string, TFrontendEngineFieldSchema>;
```

Comment:
need to inherit custom validation rules
```suggestion
	fieldSchema: Record<string, TFrontendEngineFieldSchema<V>>;
```

Reply:
quick check, I recall that custom rules added through `addCustomValidation` are tied to a specific FEE instance based on the `yupId`. do they get inherited by a nested FEE instance?

Reply:
good point, I dont think it does because the `yupId` will be unique. there is a need to align the `yupId` in nested instances, not sure if it's possible to check for a parent context or find a way to pass to the nested FEE instance

Reply:
it should be possible, I'm thinking either
- FEE to accept `yupId` as prop
- FEE to internally track its parent(s) and do a mapping. this one will be more complex

would propose to tackle this when the `unique` rule is added?